plugins/repository-s3/src/test/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStoreContainerTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.aws.blobstore;
+
+import com.amazonaws.Protocol;
+import com.amazonaws.services.s3.AmazonS3;
+import org.elasticsearch.cloud.aws.InternalAwsS3Service;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public class S3BlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+    protected BlobStore newBlobStore() throws IOException {
+        int maxRetries = 5;
+        String region = null;
+        Settings settings = Settings.EMPTY;
+
+        InternalAwsS3Service service = new InternalAwsS3Service(settings);
+        AmazonS3 client = service.client(null, Protocol.HTTPS, region, null, null, maxRetries);
+        String bucket = randomAsciiOfLength(randomIntBetween(1, 10)).toLowerCase(Locale.ROOT);
+
+        return new S3BlobStore(settings, client, bucket, region, false,
+            new ByteSizeValue(100), maxRetries, "public-read-write", "standard");
+    }
```

Comment:
@abeyad I think the idea behind ESBlobStoreContainerTestCase was to test the actual (not mocked, unless we can guarantee that mock works exactly like the real store) behavior of the store on the real storage. We cannot run it always, but it should be possible to run it from time to time in a special CI instance to make sure, for example, the behavior of the actual store or its client didn't change on us. To prevent these test from running under normal condition we annotate them with `@ESIntegTestCase.ThirdParty` and provide special configuration to run them